newshub.co.nz.js

```diff
@@ -0,0 +1,219 @@
+	"creator": "Philipp Zumstein",
+	"target": "^https?://www\\.newshub\\.co\\.nz/",
+	"minVersion": "3.0",
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "gcsibv",
+	"lastUpdated": "2016-08-28 17:34:55"
+}
+
+/*
+	***** BEGIN LICENSE BLOCK *****
+	
+	Copyright © 2016 Philipp Zumstein
+	
+	This file is part of Zotero.
+	
+	Zotero is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+	
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU Affero General Public License for more details.
+	
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+	
+	***** END LICENSE BLOCK *****
+*/
+
+
+function detectWeb(doc, url) {
+	if (getSearchResults(doc, true)) {
+		return "multiple";
+	} else if (ZU.xpathText(doc, '//div[@id="article"]')) {
+		return "newspaperArticle";
+	}
+}
+
+
+function getSearchResults(doc, checkOnly) {
+	var items = {};
+	var found = false;
+	var rows = ZU.xpath(doc, '//ul[@id="searchResult" or @id="tagsArticles"]//h3/a|//a[descendant::div[@class="article-title"]]');
+	for (var i=0; i<rows.length; i++) {
+		var href = rows[i].href;
+		var title = ZU.trimInternal(rows[i].textContent);
+		if (!href || !title) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[href] = title;
+	}
+	return found ? items : false;
+}
+
+
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) == "multiple") {
+		Zotero.selectItems(getSearchResults(doc, false), function (items) {
+			if (!items) {
+				return true;
+			}
+			var articles = [];
+			for (var i in items) {
+				articles.push(i);
+			}
+			ZU.processDocuments(articles, scrape);
+		});
+	} else {
+		scrape(doc, url);
+	}
+}
+
+
+function scrape(doc, url) {
+	var translator = Zotero.loadTranslator('web');
+	// Embedded Metadata
+	translator.setTranslator('951c027d-74ac-47d4-a107-9c3069ab7b48');
+	//translator.setDocument(doc);
+
+	translator.setHandler('itemDone', function (obj, item) {
+		
+		var author = ZU.xpathText(doc, '//div[@id="byline_author"]//span[@itemprop="name"]') || ZU.xpathText(doc, '//div[@id="article_start"]/p/strong[contains(text(), "By")]');
+		if (author) {
+			author = author.replace("By", '');
+			item.creators.push(ZU.cleanAuthor(author, "author"));
+		}
+		
+		item.date = ZU.strToISO(ZU.xpathText(doc, '//div[@id="byline_date"]'));
+		
+		var tags = ZU.xpath(doc, '//ul[@id="relatedTags"]/li/a');
+		for (var i=0; i<tags.length; i++) {
+			item.tags.push(tags[i].textContent);
+		}
+		
+		item.complete();
+	});
+
+	translator.getTranslatorObject(function(trans) {
+		trans.itemType = "newspaperArticle";
```

Comment:
I wonder if newspaper article is the best item type here? Since this isn't a newspaper but rather a website for a TV station (right?) website might make more sense?
Either way, we should list Newshub (or Newshub.co.nz ?) as the publication title -- be it as `publicationTitle` for newspaper or as `websiteTitle` for the webpage.